README.md

```diff
@@ -1,168 +1,177 @@
+In particular, the [triggers](./triggers.md) and [opt-in](./opt-in.md) can be designed in a generic way, so that they can also be used to trigger and opt-in to prefetching (of documents, in particular). Similarly, the [prerendering fetching infrastructure](./fetch.md) will likely be used for modernized prefetching.
 
-If approaches which restrict access to credentials and storage are used, __existing content may be broken__ if prefetched or prerendered in this way. This could be mitigated by user agent heuristics to only load content believed to work correctly (e.g. via reporting from past visitors to that site or from a robot indexer) or by an author declaration whether the resource will function correctly with this modified behavior. This document proposes the latter.
+## Example scenarios
 
-Server behavior in response to a GET request __may not be idempotent__. Thus issuing a request early may lead to changes in server state, such as modifying data or recording an ad impression. Unauthenticated requests to servers on the public Internet are believed to be at somewhat lower risk, because the lack of credential likely does not convey significant authority to make destructive changes and because such servers already contend with web crawlers which issue such requests. In the same-origin case, an express signal from the author that the user agent should prerender is also an indication that the request is likely to be idempotent. An alternative future direction for exploration is a capability for an origin to declare that certain URLs can be accessed idempotently, for example by responding to OPTIONS requests or by serving a manifest from a well-known URL, but this suggests a higher barrier to entry.
+### Same-origin drop-in speedup
 
-There is a risk that loading the resource may be __resource-intensive__. This is largely out of scope of this document. User agents may cancel prerendering if it has become too resource-intensive and may apply a variety of heuristics to make an educated decision about whether prerendering is likely to be lightweight.
+One of the simplest things a web developer can do is indicate that their site is prepared for the browser to prerender most or all of its content. They would put something like the following in their `<head>`:
 
-Finally, there is the risk that any proposal is __too arduous for adoption__. Unless the effect on author incentives is extremely strong it will be very difficult to convince them to deploy a complex solution. The simpler and easier to adopt, the more likely adoption will become widespread.
-
-## Alternatives considered
+```html
+<script type="speculationrules">
+{
+  "allow": [
+    {
+      "action": "prerender",
+      "url_patterns": ["/**"]
+    }
+  ],
+  "disallow": [
+    {
+      "action": "prefetch",
```

Comment:
Can "action" have multiple actions? (Related to the resolved comment thread, but I'm a bit concerned about introducing ordering or inclusion relationships between multiple actions)

Reply:
It could (though alternatively you could just repeat the whole rule).

I would really like to keep some form of ordering because I really think we should be able to prefetch but not prerender if we have network bandwidth available but don't want to waste CPU/memory, or if we're uncertain between various possible outgoing links. It's definitely possible to make this less footgun-y.

Reply:
Sure, ordering is a hard problem in general.  In my ideal mental model prerendering can be decomposed into { connect, fetch-main, parse-html, fetch-sub, run-javascripts, render ... } or something, but it may be too detailed.

Reply:
Yeah, @domenic and I are discussing some of the options here in the internal doc (which will be probably published to this repo shortly).